hrabsence/tests/phpunit/CRM/HRAbsence/BAO/HRAbsenceTypeTest.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+use Civi\Test\HeadlessInterface;
+use Civi\Test\TransactionalInterface;
+use CRM_HRCore_Test_Fabricator_Contact as ContactFabricator;
+use CRM_HRAbsence_Test_Fabricator_HRAbsenceType as AbsenceTypeFabricator;
+use CRM_Hrjobcontract_Test_Fabricator_HRJobContract as HRJobContractFabricator;
+use CRM_Hrjobcontract_Test_Fabricator_HRJobLeave as LeaveFabricator;
+use CRM_HRCore_Test_Fabricator_Activity as ActivityFabricator;
+
+/**
+ * Class CRM_HRAbsence_BAO_HRAbsenceTypeTest
+ * Runs tests on AbsenceType BAO Class
+ * 
+ * @group headless
+ */
+class CRM_HRAbsence_BAO_HRAbsenceTypeTest extends PHPUnit_Framework_TestCase 
+  implements HeadlessInterface, TransactionalInterface {
+  
+  protected $absenceType;
+  protected $contact;
+  protected $jobContract;
+  protected $leave;
+  
+  public function setUpHeadless() {
+    return \Civi\Test::headless()
+      ->install('uk.co.compucorp.civicrm.hrcore')
+      ->install('org.civicrm.hrjobcontract')
+      ->installMe(__DIR__)
+      ->apply();
+  }
+    
+  protected function setUp() {
+    // Create Absence Type
+    $this->absenceType = AbsenceTypeFabricator::fabricateUsingAPI();
+    
+    // Create Contact, Contract and Leave for Absence Type
+    $this->contact = ContactFabricator::fabricate(['first_name' => 'john', 'last_name' => 'shepard']);

```

Comment:
Tip: if you're not going to use the name of the contact anywhere on the test, you can just call the `fabricate()` method without params